celery/backends/base.py

```diff
@@ -655,11 +655,36 @@ def _forget(self, task_id):
 
     def _store_result(self, task_id, result, state,
                       traceback=None, request=None, **kwargs):
+
+        if state in self.READY_STATES:
+            date_done = datetime.datetime.utcnow()
+        else:
+            date_done = None
+
         meta = {
-            'status': state, 'result': result, 'traceback': traceback,
+            'status': state,
+            'result': result,
+            'traceback': traceback,
             'children': self.current_task_children(request),
             'task_id': bytes_to_str(task_id),
+            'date_done': date_done,
         }
+
+        if request:
+            request_meta = {
+                # Some unit tests mock the properties, casting to string to
+                # avoid serialization errors
+                'name': str(getattr(request, 'task', None)),
+                'args': getattr(request, 'args', None),
```

Comment:
I wonder if there is any security concern with storing arguments. I am afraid we may need to allow users to opt-out from this. What do you think @johnarnold ?

Reply:
@georgepsarakis Hmm... interesting question. I think if you have access the the results backend, you're already within the security boundary.

We also store the args in the broker when we serialize the task, and probably >80% of users are using the same redis server for both broker/results :)

We also include the args in the (send-task?) event / are  visible in flower.

I can certainly/easily add an option to disable, but would like to give others a chance to comment?

Reply:
@johnarnold sure, let's wait to get other opinions on this. There are also some storage considerations along with the change (think of a large JSON payload. e.g from an API call, as argument being passed to the task). Broker messages are ephemeral, but results may be desired to be retained for longer time in some occasions.

I am inclined to have the setting and a warning in the logs that this behavior will be the default from the next minor version onward. If that sounds like too much work, I can help with the implementation.

@thedrow @auvipy would like to share your opinion perhaps?

Reply:
@georgepsarakis I added a result_extended option, defaulted to True.  Does that look acceptable?